pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/uuid/GetAllUUIDMetadata.ios.kt

```diff
@@ -54,27 +54,29 @@ class GetAllUUIDMetadataImpl(
             },
             includeCount = includeCount,
             includeCustom = includeCustom,
-            onSuccess = callback.onSuccessHandler {
-                PNUUIDMetadataArrayResult(
-                    status = it?.status()?.toInt() ?: 0,
-                    data = (it?.data() as List<PubNubUUIDMetadataObjC>).map { metadata ->
-                        PNUUIDMetadata(
-                            id = metadata.id(),
-                            name = metadata.name(),
-                            externalId = metadata.externalId(),
-                            profileUrl = metadata.profileUrl(),
-                            email = metadata.email(),
-                            custom = metadata.custom() as? Map<String, Any?>,
-                            updated = metadata.updated(),
-                            eTag = metadata.eTag(),
-                            type = metadata.type(),
-                            status = metadata.status()
-                        )
-                    },
-                    totalCount = it.totalCount()?.intValue ?: 0,
-                    next = it.next()?.end()?.let { hash -> PNPage.PNNext(pageHash = hash) },
-                    prev = it.next()?.start()?.let { hash -> PNPage.PNPrev(pageHash = hash) }
-                )
+            onSuccess = { data, totalCount, next ->
```

Comment:
I removed Objective-C class like `PubNubGetChannelMetadataResultObjC` and moved all parameters in the closure instead. Is it a better approach or not?

Reply:
if it makes the code simpler (and maybe smaller - one less class?) then sure, it's not to be used by anyone except us internally, so whatever you prefer here

Reply:
I introduced a method like `onSuccessHandler3` we talked about today.
Let me know if you prefer a better name for that